configuration/windows-gaming-hvm.md

```diff
@@ -0,0 +1,163 @@
+---
+layout: doc
+title: Windows gaming hvm
+permalink: /doc/windows-gaming-hvm/
+---
+
+# Windows gaming HVM
+
+Some information to configure a windows HVM for gaming. 
+This is not officially supported, just some community trial & errors
+
+## References
+
+Everythings needed is referenced here
+
+- [Usefull technical details](https://paste.debian.net/1043341/)
+- [Reddit thread of what is needed for GPU passthrough](https://www.reddit.com/r/Qubes/comments/9hp3e7/gpu_passthrough_howto/)
+- [Solution to have more than 3Go of RAM in the Windows HVM](https://github.com/QubesOS/qubes-issues/issues/4321#issuecomment-423011787)
+- [Some old references](https://www.reddit.com/r/Qubes/comments/66wk4q/gpu_passthrough/)
+
+## Prerequise
+
+You have a functional Windows 7 HVM.
+
+The "how to" for this part can be found on the Qubes OS documentation and here: [Usefull github comment](https://github.com/QubesOS/qubes-issues/issues/3585#issuecomment-453200971).
+
+However, few tips:
+
+- Do a backup (clone VM) of the Windows HVM BEFORE starting to install QWT
+- The Windows user MUST BE "user"
+- Windows 7 Only, do not use Windows 10 or others.
+
+## Hardware
+
+To have a Windows HVM for gaming, you must have:
+
+- A dedicated AMD GPU. By dedicated, it means: it is a secondary GPU, not the GPU used to display dom0. Nvidia GPU are not supported (or maybe with a lot of trick).
+- A really fast disk (M.2 disk)
+- A lot of RAM
+- A dedicated screen
+
+In my case, I use:
+
+- Secondary GPU: AMD RX580
+- Primary GPU: Some Nvidia trash, used for dom0
+- 32Go of RAM. 16Go of RAM will be dedicated for the Windows HVM
+- A fast M.2 disk
+
+
+## Checklist
+
+Short list of things to do to make the GPU passthrough work:
+
+- In dom0, you edited the file /etc/default/grub to allow PCI hidding for your secondary GPU, and regenerated the grub
+- You patched your stubdom-linux-rootfs.gz to allow to have more than 3Go of RAMfor your Windows HVM
+
+## GRUB modification
+
+You must hide your secondary GPU from dom0.
+To do that, you have to edit the GRUB.
+
+In a dom0 Terminal, type:
+
+```bash
+qvm-pci
+```
+
+Then find the devices id for your secondary gpu.
+
+In my case, it is "dom0:0a_00.0" and "dom0:0a_00.1".
+
+Edit /etc/default/grub, and add the PCI hiding 
+
+```
+GRUB_CMDLINE_LINUX="....rd.qubes.hide_pci=0a:00.0,0a:00.1 modprobe=xen-pciback.passthrough=1 xen-pciback.permissive"
```

Comment:
`modprobe=xen-pciback.passthrough=1` looks weird, I think it does nothing.
`xen-pciback.permissive` - better to enable it for a particular device. You can do that with qvm-pci:
```
qvm-pci attach windows-hvm dom0:0a_00.0 -o permissive=True -o no-strict-reset=True
```

Reply:
Hi, I just checked, 

```
modprobe=xen-pciback.passthrough=1
```

was indeed useless. 
And the ``qvm-pci ... -o permissive=True`` replace correctly the ``xen-pciback.permissive``. 

But in fact, the "permissive" doesn't seems to be necessary at all, still running few tests, but seems like it is not required

I will update my doc & send it to Qube Community docs. 

Also:
- There is a mistake in the official qubes doc https://www.qubes-os.org/doc/pci-devices/#permissive doesn't contain the "permissive=True" 
- Does a doc with the full list of all the options we can pass ``-o xxxx`` exist ? 

Thanks ! :)

Reply:
Who broke my document, it used to have the right setting in there! :) Anyways, `man qvm-device` describes the available options.

Reply:
Submitted https://github.com/QubesOS/qubes-doc/pull/812 to fix the example.

Reply:
@awokd thanks :) 

And I confirm, all the "permissive" flag are unnecessary for this usecase